client/manager_redis.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"encoding/json"
+
+	"github.com/imdario/mergo"
+	"github.com/ory-am/fosite"
+	"github.com/ory-am/hydra/pkg"
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
+	"gopkg.in/redis.v5"
+)
+
+type RedisManager struct {
+	DB        *redis.Client
+	Hasher    fosite.Hasher
+	KeyPrefix string
+}
+
+const redisClientTemplate = "hydra:clients"
+
+func (m *RedisManager) redisClientKey() string {
+	return m.KeyPrefix + redisClientTemplate
+}
+
+func (m *RedisManager) GetConcreteClient(id string) (*Client, error) {
+	resp, err := m.DB.HGet(m.redisClientKey(), id).Bytes()
+	if err == redis.Nil {
+		return nil, errors.Wrap(pkg.ErrNotFound, "")
+	} else if err != nil {
+		return nil, errors.Wrap(err, "")
+	}
+
+	var d *Client
+	if err := json.Unmarshal(resp, &d); err != nil {
```

Comment:
This can't work. You're referencing a nil-Pointer. This should be something like this instead:

```
var d Client
if err := json.Unmarshal(resp, &d); err != nil {
    return nil, errors.Wrap(err, "")
}
return &d, nil
```

I don't know if you've seen it, but I think the Go tour gives good ideas how this works: https://tour.golang.org/moretypes/1